diff -urN OpenFOAM-2006.orig/src/OpenFOAM/db/IOstreams/hashes/OSHA1stream.H OpenFOAM-2006/src/OpenFOAM/db/IOstreams/hashes/OSHA1stream.H
--- OpenFOAM-2006.orig/src/OpenFOAM/db/IOstreams/hashes/OSHA1stream.H	2020-06-29 17:16:22.000000000 +0000
+++ OpenFOAM-2006/src/OpenFOAM/db/IOstreams/hashes/OSHA1stream.H	2024-05-22 06:55:53.000000000 +0000
@@ -32,8 +32,8 @@
 
 \*---------------------------------------------------------------------------*/
 
-#ifndef OSHA1stream_H
-#define OSHA1stream_H
+#ifndef Foam_OSHA1stream_H
+#define Foam_OSHA1stream_H
 
 #include "OSstream.H"
 #include "SHA1.H"
@@ -63,10 +63,17 @@
 
     protected:
 
+        //- Handle overflow
+        virtual int overflow(int c = EOF)
+        {
+            if (c != EOF) sha1_.append(c);
+            return c;
+        }
+
         //- Put sequence of characters
         virtual std::streamsize xsputn(const char* s, std::streamsize n)
         {
-            sha1_.append(s, n);
+	    if (n) sha1_.append(s, n);
             return n;
         }
 
diff -urN OpenFOAM-2006.orig/src/OpenFOAM/primitives/hashes/SHA1/SHA1.H OpenFOAM-2006/src/OpenFOAM/primitives/hashes/SHA1/SHA1.H
--- OpenFOAM-2006.orig/src/OpenFOAM/primitives/hashes/SHA1/SHA1.H	2020-06-29 17:16:22.000000000 +0000
+++ OpenFOAM-2006/src/OpenFOAM/primitives/hashes/SHA1/SHA1.H	2024-05-22 06:54:07.000000000 +0000
@@ -42,8 +42,8 @@
 
 \*---------------------------------------------------------------------------*/
 
-#ifndef SHA1_H
-#define SHA1_H
+#ifndef Foam_SHA1_H
+#define Foam_SHA1_H
 
 #include <string>
 #include <cstdint>
@@ -113,6 +113,9 @@
         //- Reset the hashed data before appending more
         void clear();
 
+	//- Append single character
+	inline void append(char c);
+
         //- Append data for processing
         inline SHA1& append(const char* str);
 
diff -urN OpenFOAM-2006.orig/src/OpenFOAM/primitives/hashes/SHA1/SHA1I.H OpenFOAM-2006/src/OpenFOAM/primitives/hashes/SHA1/SHA1I.H
--- OpenFOAM-2006.orig/src/OpenFOAM/primitives/hashes/SHA1/SHA1I.H	2020-06-29 17:16:22.000000000 +0000
+++ OpenFOAM-2006/src/OpenFOAM/primitives/hashes/SHA1/SHA1I.H	2024-05-22 06:54:07.000000000 +0000
@@ -53,6 +53,11 @@
 
 // * * * * * * * * * * * * * * * Member Functions  * * * * * * * * * * * * * //
 
+inline void Foam::SHA1::append(char c)
+{
+	    processBytes(&c, 1);
+}
+
 inline Foam::SHA1& Foam::SHA1::append(const char* data, size_t len)
 {
     processBytes(data, len);
