diff -urN ThirdParty-v2006.orig/ADIOS2-2.4.0/thirdparty/dill/dill/arm8.c ThirdParty-v2006/ADIOS2-2.4.0/thirdparty/dill/dill/arm8.c
--- ThirdParty-v2006.orig/ADIOS2-2.4.0/thirdparty/dill/dill/arm8.c	2019-06-30 20:01:38.000000000 +0000
+++ ThirdParty-v2006/ADIOS2-2.4.0/thirdparty/dill/dill/arm8.c	2024-05-19 16:30:53.000000000 +0000
@@ -1517,6 +1517,8 @@
     }
 }
 
+extern void arm8_rt_call_link(char *code, call_t *t);
+
 static void
 arm8_call_link(dill_stream s)
 {
@@ -1539,7 +1541,9 @@
  *  Cache flush code grabbed from a Dec 1999 posting on libc-hacker 
  *  mailing list
  */
-extern void __clear_cache(char*, char *);
+#ifndef CLEAR_CACHE_DEFINED
+extern void __clear_cache(void *, void *);
+#endif
 
 static void
 arm8_flush(void *base, void *limit)
diff -urN ThirdParty-v2006.orig/ADIOS2-2.4.0/thirdparty/dill/dill/arm8_rt.c ThirdParty-v2006/ADIOS2-2.4.0/thirdparty/dill/dill/arm8_rt.c
--- ThirdParty-v2006.orig/ADIOS2-2.4.0/thirdparty/dill/dill/arm8_rt.c	2019-06-30 20:01:38.000000000 +0000
+++ ThirdParty-v2006/ADIOS2-2.4.0/thirdparty/dill/dill/arm8_rt.c	2024-05-19 16:30:37.000000000 +0000
@@ -124,7 +124,9 @@
  *  Cache flush code grabbed from a Dec 1999 posting on libc-hacker 
  *  mailing list
  */
-extern void __clear_cache(char*, char *);
+#ifndef CLEAR_CACHE_DEFINED
+extern void __clear_cache(void *, void *);
+#endif
 
 static void
 arm8_flush(void *base, void *limit)
