diff -urN WRF-4.x.orig/arch/configure.defaults WRF-4.x/arch/configure.defaults
--- WRF-4.x.orig/arch/configure.defaults	2021-01-15 17:21:58.000000000 +0000
+++ WRF-4.x/arch/configure.defaults	2022-11-14 23:50:46.658836696 +0000
@@ -757,18 +757,18 @@
 CC_TOOLS        =      $(SCC) 
 
 ###########################################################
-#ARCH    Linux x86_64 ppc64le, gfortran compiler with gcc  #serial smpar dmpar dm+sm
+#ARCH    Linux x86_64 ppc64le, flang compiler with clang  #serial smpar dmpar dm+sm
 #
-DESCRIPTION     =       GNU ($SFC/$SCC)
+DESCRIPTION     =       BSD ($SFC/$SCC)
 DMPARALLEL      =       # 1
 OMPCPP          =       # -D_OPENMP
 OMP             =       # -fopenmp
 OMPCC           =       # -fopenmp
-SFC             =       gfortran
-SCC             =       gcc
-CCOMP           =       gcc
+SFC             =       flang
+SCC             =       clang
+CCOMP           =       clang
 DM_FC           =       mpif90 -f90=$(SFC)
-DM_CC           =       mpicc -cc=$(SCC)
+DM_CC           =       mpicc -DMPI2_SUPPORT
 FC              =       CONFIGURE_FC
 CC              =       CONFIGURE_CC
 LD              =       $(FC)
@@ -793,10 +793,10 @@
 MODULE_SRCH_FLAG =     
 TRADFLAG        =      CONFIGURE_TRADFLAG
 CPP             =      /lib/cpp CONFIGURE_CPPFLAGS
-AR              =      ar
+AR              =      llvm-ar
 ARFLAGS         =      ru
 M4              =      m4 -G
-RANLIB          =      ranlib
+RANLIB          =      llvm-ranlib
 RLFLAGS		=	
 CC_TOOLS        =      $(SCC) 
 
diff -urN WRF-4.x.orig/configure WRF-4.x/configure
--- WRF-4.x.orig/configure	2021-01-15 17:21:58.000000000 +0000
+++ WRF-4.x/configure	2022-11-14 23:49:51.673961645 +0000
@@ -686,7 +686,7 @@
   foo=foo_$$
 
   grep '^SFC' configure.wrf | grep -i 'gfortran' > /dev/null
-  if [ $? ]
+  if [ $? -eq 0 ]
   then
 
     cat > ${foo}.F << EOF
