diff -urN WRF-3.w.orig/arch/configure_new.defaults WRF-3.w/arch/configure_new.defaults
--- WRF-3.w.orig/arch/configure_new.defaults	2015-06-26 21:59:10.000000000 +0000
+++ WRF-3.w/arch/configure_new.defaults	2023-03-17 12:40:20.000000000 +0000
@@ -43,7 +43,7 @@
 CC_TOOLS        =      cc 
 
 ###########################################################
-#ARCH    Linux i486 i586 i686, gfortran compiler with gcc #serial smpar dmpar dm+sm
+#ARCH   Linux   armv7l aarch64, gnu OpenMPI #serial smpar dmpar dm+sm
 #
 DESCRIPTION     =       GNU ($SFC/$SCC)
 DMPARALLEL      =       # 1
@@ -65,12 +65,12 @@
 LDFLAGS_LOCAL   =
 CPLUSPLUSLIB    =       
 ESMF_LDFLAG     =       $(CPLUSPLUSLIB)
-FCOPTIM         =       -O2 -ftree-vectorize -funroll-loops
+FCOPTIM         =       -Ofast -mcpu=native -ftree-vectorize -funroll-loops -fallow-argument-mismatch -fallow-invalid-boz
 FCREDUCEDOPT	=       $(FCOPTIM)
 FCNOOPT         =       -O0
 FCDEBUG         =       # -g $(FCNOOPT)  # -fbacktrace -ggdb -fbounds-check -ffpe-trap=invalid,zero,overflow
-FORMAT_FIXED    =       -ffixed-form
-FORMAT_FREE     =       -ffree-form -ffree-line-length-none
+FORMAT_FIXED    =       -ffixed-form 
+FORMAT_FREE     =       -ffree-form -ffree-line-length-none -fallow-argument-mismatch -fallow-invalid-boz
 FCSUFFIX        =       
 BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO)
