diff -urN WRF-4.x.orig/arch/configure.defaults WRF-4.x/arch/configure.defaults
--- WRF-4.x.orig/arch/configure.defaults	2018-11-10 07:55:46.000000000 +0800
+++ WRF-4.x/arch/configure.defaults	2023-03-15 14:41:42.541020451 +0800
@@ -296,13 +296,13 @@
 DESCRIPTION     =       INTEL ($SFC/$SCC)
 DMPARALLEL      =       # 1
 OMPCPP          =       # -D_OPENMP
-OMP             =       # -openmp -fpp -auto
-OMPCC           =       # -openmp -fpp -auto
+OMP             =       # -qopenmp -fpp -auto
+OMPCC           =       # -qopenmp -fpp -auto
 SFC             =       ifort
 SCC             =       icc
 CCOMP           =       icc
-DM_FC           =       mpif90 -f90=$(SFC)
-DM_CC           =       mpicc -cc=$(SCC)
+DM_FC           =       mpiifort
+DM_CC           =       mpiicc
 FC              =       CONFIGURE_FC
 CC              =       CONFIGURE_CC
 LD              =       $(FC)
