diff -urN WRF-4.x.orig/arch/configure.defaults WRF-4.x/arch/configure.defaults
--- WRF-4.x.orig/arch/configure.defaults	2021-01-15 17:21:58.000000000 +0000
+++ WRF-4.x/arch/configure.defaults	2023-09-07 01:19:32.000000000 +0000
@@ -757,46 +757,54 @@
 CC_TOOLS        =      $(SCC) 
 
 ###########################################################
-#ARCH    Linux x86_64 ppc64le, gfortran compiler with gcc  #serial smpar dmpar dm+sm
+#ARCH    AMD Linux x86_64, AOCC flang compiler with AOCC clang  #serial smpar dmpar dm+sm
+# Supported AMDARCH are znver1, znver2 and znver3 for ZEN1, ZEN2 and ZEN3 respectively
+# For optimized AMDFCFLAGS and AMDLDFLAGS, please reach out to toolchainsupport@amd.com
 #
-DESCRIPTION     =       GNU ($SFC/$SCC)
+DESCRIPTION     =       AMD ($SFC/$SCC) :  AMD ZEN1/ ZEN2/ ZEN3 Architectures
 DMPARALLEL      =       # 1
 OMPCPP          =       # -D_OPENMP
 OMP             =       # -fopenmp
 OMPCC           =       # -fopenmp
-SFC             =       gfortran
-SCC             =       gcc
-CCOMP           =       gcc
+SFC             =       flang
+SCC             =       clang
+CCOMP           =       clang
 DM_FC           =       mpif90 -f90=$(SFC)
 DM_CC           =       mpicc -cc=$(SCC)
 FC              =       CONFIGURE_FC
 CC              =       CONFIGURE_CC
 LD              =       $(FC)
 RWORDSIZE       =       CONFIGURE_RWORDSIZE
+
+AMDARCH         =       -march=znver3
+AMDMATHLIB      =       -fveclib=AMDLIBM
+AMDLDFLAGS      =       -Wl,-mllvm -Wl,-enable-loop-reversal -Wl,-mllvm -Wl,-enable-gather -Wl,-mllvm -Wl,-vectorize-non-contiguous-memory-aggressively
+AMDFCFLAGS      =
+
 PROMOTION       =       #-fdefault-real-8
 ARCH_LOCAL      =       -DNONSTANDARD_SYSTEM_SUBR  -DWRF_USE_CLM
-CFLAGS_LOCAL    =       -w -O3 -c  # -DRSL0_ONLY
-LDFLAGS_LOCAL   =       
+CFLAGS_LOCAL    =       -w -c -m64 -Ofast -ffast-math $(AMDARCH)
+LDFLAGS_LOCAL   =       -m64 -Ofast -Mstack_arrays $(AMDARCH) $(AMDLDFLAGS) $(AMDMATHLIB) -lamdlibm -lomp -ltirpc
 CPLUSPLUSLIB    =       
 ESMF_LDFLAG     =       $(CPLUSPLUSLIB)
-FCOPTIM         =       -O2 -ftree-vectorize -funroll-loops
-FCREDUCEDOPT	=       $(FCOPTIM)
-FCNOOPT		=       -O0
+FCOPTIM         =       -Ofast -ffast-math $(AMDARCH) -Mbyteswapio -Mstack_arrays -ftree-vectorize -Mbyteswapio -funroll-loops -finline-aggressive -finline-hint-functions $(AMDMATHLIB) $(AMDFCFLAGS)
+FCREDUCEDOPT    =       -O3 $(AMDARCH) -ffast-math -Mstack_arrays -DFCREDUCEDOPT
+FCNOOPT         =       -O0
 FCDEBUG         =       # -g $(FCNOOPT) # -ggdb -fbacktrace -fcheck=bounds,do,mem,pointer -ffpe-trap=invalid,zero,overflow
-FORMAT_FIXED    =       -ffixed-form
-FORMAT_FREE     =       -ffree-form -ffree-line-length-none
+FORMAT_FIXED    =       -Mfixed
+FORMAT_FREE     =       -Mfreeform
 FCSUFFIX        =       
 FCCOMPAT        =       
-BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4
-FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO) $(FCCOMPAT)
-FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
+BYTESWAPIO      =       -Mbyteswapio
+FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO)
+FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG) -DBASEOPTS -fopenmp
 MODULE_SRCH_FLAG =     
-TRADFLAG        =      CONFIGURE_TRADFLAG
-CPP             =      /lib/cpp CONFIGURE_CPPFLAGS
-AR              =      ar
+TRADFLAG        =      -traditional
+CPP             =      /lib/cpp -P
+AR              =      llvm-ar
 ARFLAGS         =      ru
 M4              =      m4 -G
-RANLIB          =      ranlib
+RANLIB          =      llvm-ranlib
 RLFLAGS		=	
 CC_TOOLS        =      $(SCC) 
 
diff -urN WRF-4.x.orig/configure WRF-4.x/configure
--- WRF-4.x.orig/configure	2021-01-15 17:21:58.000000000 +0000
+++ WRF-4.x/configure	2023-09-07 01:06:27.000000000 +0000
@@ -686,7 +686,7 @@
   foo=foo_$$
 
   grep '^SFC' configure.wrf | grep -i 'gfortran' > /dev/null
-  if [ $? ]
+  if [ $? -eq 0 ]
   then
 
     cat > ${foo}.F << EOF
