diff -urN WRF-4.z.orig/arch/configure.defaults WRF-4.z/arch/configure.defaults
--- WRF-4.z.orig/arch/configure.defaults	2022-08-25 22:42:53.000000000 +0000
+++ WRF-4.z/arch/configure.defaults	2022-11-25 13:00:36.000000000 +0000
@@ -67,7 +67,7 @@
 LDFLAGS_LOCAL   =
 CPLUSPLUSLIB    =       
 ESMF_LDFLAG     =       $(CPLUSPLUSLIB)
-FCOPTIM         =       -O2 -ftree-vectorize -funroll-loops
+FCOPTIM         =       -O3 -ftree-vectorize -funroll-loops -march=native
 FCREDUCEDOPT	=       $(FCOPTIM)
 FCNOOPT         =       -O0
 FCDEBUG         =       # -g $(FCNOOPT)  # -fbacktrace -ggdb -fcheck=bounds,do,mem,pointer -ffpe-trap=invalid,zero,overflow
@@ -80,7 +80,7 @@
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =
 TRADFLAG        =      CONFIGURE_TRADFLAG
-CPP             =      /lib/cpp CONFIGURE_CPPFLAGS
+CPP             =      cpp CONFIGURE_CPPFLAGS
 AR              =      ar
 ARFLAGS         =      ru
 M4              =      m4 -B 14000
@@ -2057,7 +2057,7 @@
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =
 TRADFLAG        =      CONFIGURE_TRADFLAG
-CPP             =      /lib/cpp CONFIGURE_CPPFLAGS
+CPP             =      cpp CONFIGURE_CPPFLAGS
 AR              =      ar
 ARFLAGS         =      ru
 M4              =      m4 -G
