diff -urN WPS.orig/arch/configure.defaults WPS/arch/configure.defaults
--- WPS.orig/arch/configure.defaults	2022-06-16 00:35:48.302096822 +0000
+++ WPS/arch/configure.defaults	2022-06-16 00:42:35.637500599 +0000
@@ -193,6 +193,30 @@
 RANLIB              = ranlib 
 
 ########################################################################################################################
+#ARCH    Linux aarch64, flang   # serial serial_NO_GRIB2 dmpar dmpar_NO_GRIB2
+#
+COMPRESSION_LIBS    = CONFIGURE_COMP_L
+COMPRESSION_INC     = CONFIGURE_COMP_I
+FDEFS               = CONFIGURE_FDEFS
+SFC                 = flang
+SCC                 = clang
+DM_FC               = mpif90
+DM_CC               = mpicc
+FC                  = CONFIGURE_FC 
+CC                  = CONFIGURE_CC
+LD                  = $(FC)
+FFLAGS              = -ffree-form -O -fconvert=big-endian -frecord-marker=4
+F77FLAGS            = -ffixed-form -O -fconvert=big-endian -frecord-marker=4
+FCCOMPAT            = CONFIGURE_COMPAT_FLAGS
+FCSUFFIX            = 
+FNGFLAGS            = $(FFLAGS)
+LDFLAGS             =
+CFLAGS              =
+CPP                 = /usr/bin/cpp -P -traditional
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DBIT32 -DNO_SIGNAL CONFIGURE_MPI
+RANLIB              = armllvm-ranlib 
+
+########################################################################################################################
 #ARCH    Linux x86_64, PGI compiler  # serial serial_NO_GRIB2 dmpar dmpar_NO_GRIB2
 #
 COMPRESSION_LIBS    = CONFIGURE_COMP_L
diff -urN WPS.orig/arch/preamble WPS/arch/preamble
--- WPS.orig/arch/preamble	2022-06-16 00:53:18.391528778 +0000
+++ WPS/arch/preamble	2022-06-16 00:47:20.897496965 +0000
@@ -15,7 +15,7 @@
 
 SHELL           	=       /bin/sh
 
-NCARG_LIBS		=	-L$(NCARG_ROOT)/lib -lncarg -lncarg_gks -lncarg_c \
+NCARG_LIBS		=	-L$(NCARG_ROOT)/lib -L$(NCARG_ROOT)/lib64 -lncarg -lncarg_gks -lncarg_c \
 				-lX11 -lXext -lpng -lz -lcairo -lfontconfig -lpixman-1 \
 				-lfreetype -lexpat -lpthread -lbz2 -lXrender -lgfortran -lgcc
 
@@ -45,7 +45,7 @@
                         -L$(WRF_DIR)/external/io_grib_share -lio_grib_share \
                         -L$(WRF_DIR)/external/io_int -lwrfio_int \
                         -L$(WRF_DIR)/external/io_netcdf -lwrfio_nf \
-                        -L$(NETCDF)/lib CONFIGURE_NETCDFF_LIB -lnetcdf
+                        -L$(NETCDF)/lib -L$(NETCDF)/lib64 CONFIGURE_NETCDFF_LIB -lnetcdf -lnetcdff -lgomp
 
 #### Architecture specific settings ####
 
