diff --color -urN WPS-4.x.orig/ungrib/src/ngl/g2/intmath.f WPS-4.x/ungrib/src/ngl/g2/intmath.f
--- WPS-4.x.orig/ungrib/src/ngl/g2/intmath.f	2019-01-26 07:38:00.000000000 +0800
+++ WPS-4.x/ungrib/src/ngl/g2/intmath.f	2022-10-06 23:06:23.206169845 +0800
@@ -169,7 +169,9 @@
       ilog2_2=0
       i=i_in
       if(i<=0) return
-      if(iand(i,i-1)/=0) then
+! WPS modification for the XL compiler
+!      if(iand(i,i-1)/=0) then
+      if(iand(i,i-1_2)/=0) then
          !write(0,*) 'iand i-1'
          ilog2_2=1
       endif
@@ -204,7 +206,9 @@
       ilog2_1=0
       i=i_in
       if(i<=0) return
-      if(iand(i,i-1)/=0) then
+! WPS modification for the XL compiler
+!      if(iand(i,i-1)/=0) then
+      if(iand(i,i-1_1)/=0) then
          !write(0,*) 'iand i-1'
          ilog2_1=1
       endif
