diff -urN WPS.orig/arch/configure.defaults WPS/arch/configure.defaults
--- WPS.orig/arch/configure.defaults	2022-06-16 00:35:48.302096822 +0000
+++ WPS/arch/configure.defaults	2022-06-16 00:42:35.637500599 +0000
@@ -193,6 +193,30 @@
 RANLIB              = ranlib 
 
 ########################################################################################################################
+#ARCH    Linux aarch64, flang   # serial serial_NO_GRIB2 dmpar dmpar_NO_GRIB2
+#
+COMPRESSION_LIBS    = CONFIGURE_COMP_L
+COMPRESSION_INC     = CONFIGURE_COMP_I
+FDEFS               = CONFIGURE_FDEFS
+SFC                 = armflang
+SCC                 = armclang
+DM_FC               = mpif90 -fopenmp
+DM_CC               = mpicc -fopenmp
+FC                  = CONFIGURE_FC 
+CC                  = CONFIGURE_CC
+LD                  = $(FC)
+FFLAGS              = -ffree-form -O -fconvert=big-endian
+F77FLAGS            = -ffixed-form -O -fconvert=big-endian
+FCCOMPAT            = CONFIGURE_COMPAT_FLAGS
+FCSUFFIX            = 
+FNGFLAGS            = $(FFLAGS)
+LDFLAGS             =
+CFLAGS              =
+CPP                 = cpp -P -traditional
+CPPFLAGS            = -D_UNDERSCORE -DBYTESWAP -DLINUX -DIO_NETCDF -DBIT32 -DNO_SIGNAL CONFIGURE_MPI
+RANLIB              = armllvm-ranlib 
+
+########################################################################################################################
 #ARCH    Linux x86_64, PGI compiler  # serial serial_NO_GRIB2 dmpar dmpar_NO_GRIB2
 #
 COMPRESSION_LIBS    = CONFIGURE_COMP_L
